src/couch_db_updater.erl

```diff
@@ -1454,3 +1468,44 @@ default_security_object(_DbName) ->
         "everyone" ->
             []
     end.
+
+% These functions rely on using the process dictionary. This is usually frowned
+% upon howver in this case it is done to avoid changing to a different server state
+% record. Once PSE (Pluggable Storage Engine) code lands this should be moved to the
+% #db{} record.
+
+schedule_timeout() ->
+    case get(idle_timer_ref) of
+        TimerRef when is_reference(TimerRef) ->
+            erlang:cancel_timer(TimerRef);
+        undefined  ->
+            ok
+    end,
+    case idle_limit() of
+        infinity ->
+            ok;
+        Timeout when is_integer(Timeout) ->
+            TRef = erlang:send_after(Timeout, self(), timeout),
```

Comment:
It was to preserve hibernate semantics. If we didn't have hibernate there for update_docs could have just been returning a timeout on gen_server replies. But because the reply is {reply, Result, State, Timeout|hibernate) we could not both hibernate and trigger a timeout. 

The timeouts in the gen_server replies are used to detect idleness.

I wonder if we just want to do a gc call instead of a hibernate and get rid of the extra timer management.